protocols/floodsub/src/layer.rs

```diff
@@ -168,40 +167,40 @@ impl<TSubstream> Floodsub<TSubstream> {
     }
 
     /// Publishes a message to the network, if we're subscribed to the topic only.
-    pub fn publish(&mut self, topic: impl Into<TopicHash>, data: impl Into<Vec<u8>>) {
+    pub fn publish(&mut self, topic: Topic, data: impl Into<Vec<u8>>) {
```

Comment:
Could you use `impl Into<Topic>` everywhere in the public API? (where there was `impl Into<TopicHash>` before)
And also add `impl From<String> for Topic` and `impl From<&str> for Topic`? This way, users can just pass a string here.